### DISCUSSION TITLE
tensorflow.python.framework.errors_impl.NotFoundError: undefined symbol: _ZN10tensorflow6StatusC1ENS_5error4CodeENS_11StringPieceE

### DISCUSSION THREAD

**Discussion by Tarakk** [deepmodeling/deepmd-kit#80] [Q&A]
Hi, 
Lately I installed the DeepMD kit. Installation went smoothly, however while performing a training I encounter the following error,
tensorflow.python.framework.errors_impl.NotFoundError: /deepmd-kit/source/build/deepmd_root=/deepmd-kit/lib/deepmd/libop_abi.so: undefined symbol: _ZN10tensorflow6StatusC1ENS_5error4CodeENS_11StringPieceE

As per the suggestion in the installation instructions I set '-DTF_GOOGLE_BIN=true' during cmake, and I use python/3.7.1. These do not help.

Is there any issue with this version of the python or am I missing something?
Any suggestions/recommendations?

Thanks,
Tarak

**Comment by njzjz** [comment]
Hi,
Which version of gcc compiler do you have? And
How do you install TensorFlow? Use `pip install tensorflow`?

**Comment by Tarakk** [comment]
Hi, 
It is gcc/6.3.0.
I have followed the installation instructions from here,
http://www.deepmd.org/download/installing-manual/

------------------------------------------------
**Install tensorflow’s C++ interface**

The tensorflow’s C++ interface will be compiled from the source code. Before starting compilation, it is may be a good idea to check the environment requirement of tensorflow by the officially tested configurations. If you do not have bazel, please install it.

Firstly get the source code of the tensorflow

cd /some/workspace
git clone https://github.com/tensorflow/tensorflow tensorflow

**Comment by njzjz** [comment]
@Tarakk Yes, but I think you should have installed TensorFlow's Python interface using commands like `pip install tensorflow`. TensorFlow's official Python packages use gcc 4.8 to compile until [1.15.0-rc1](https://github.com/tensorflow/tensorflow/releases/tag/v1.15.0-rc1). It will cause ABI incompatible if you use GCC>=5 to compile C++ interface and DeePMD-kit.

See my comment in #52 for details.

**Comment by njzjz** [comment]
More suggestion: print tf.version.COMPILER_VERSION to see what compiler version it uses.

**Comment by Tarakk** [comment]
thanks, I have successfully installed the code.
Here are the versions, 
gcc 5.2.0
tf-1.12
bazel-0.15
python-3.6.1
DeepMD-kit r0.12
